src/logic/safe/hooks/useFetchTokens.tsx

```diff
@@ -10,11 +10,12 @@ import { fetchTokens } from 'src/logic/tokens/store/actions/fetchTokens'
 import { COINS_LOCATION_REGEX, COLLECTIBLES_LOCATION_REGEX } from 'src/routes/safe/components/Balances'
 import { Dispatch } from 'src/logic/safe/store/actions/types.d'
 
-export const useFetchTokens = (safeAddress: string): void => {
+export const useFetchTokens = (safeAddress: string | undefined): void => {
```

Comment:
why do we add undefined there if this function cannot work as intended if the address is undefined?

Reply:
Because the `safeAddress selector` could return string or undefined and hooks cannot be called conditionally, the check condition must be done on the hook

Reply:
Ideally, this prop would come from top-level components to other ones, and somewhere on its way, we would perform a check whether its defined, so we render one set of components only if this address s defined. The selector is totally valid and the problem is how we're using it. Fixing it with `as` statement is ok because we know if that component is now rendered the address is defined.

Reply:
Ok so you propose instead of fixing the bug, that was caused by a refactor, do another refactor for the entire app to avoid an if inside a hook? that doesn't make sense to me, I prefer to delivery the fix

Reply:
@Agupane 
> Fixing it with as statement is ok because we know if that component is now rendered the address is defined.